include/oneapi/tbb/blocked_nd_range.h

```diff
@@ -37,45 +34,48 @@ namespace detail {
 namespace d1 {
 
 /*
-    The blocked_rangeNd_impl uses make_index_sequence<N> to automatically generate a ctor with
+    The blocked_nd_range_impl uses make_index_sequence<N> to automatically generate a ctor with
     exactly N arguments of the type tbb::blocked_range<Value>. Such ctor provides an opportunity
     to use braced-init-list parameters to initialize each dimension.
     Use of parameters, whose representation is a braced-init-list, but they're not
     std::initializer_list or a reference to one, produces a non-deduced context
     within template argument deduction.
 
-    NOTE: blocked_rangeNd must be exactly a templated alias to the blocked_rangeNd_impl
+    NOTE: blocked_nd_range must be exactly a templated alias to the blocked_nd_range_impl
     (and not e.g. a derived class), otherwise it would need to declare its own ctor
     facing the same problem that the impl class solves.
 */
 
 template<typename Value, unsigned int N, typename = detail::make_index_sequence<N>>
     __TBB_requires(blocked_range_value<Value>)
-class blocked_rangeNd_impl;
+class blocked_nd_range_impl;
 
 template<typename Value, unsigned int N, std::size_t... Is>
     __TBB_requires(blocked_range_value<Value>)
-class blocked_rangeNd_impl<Value, N, detail::index_sequence<Is...>> {
+class blocked_nd_range_impl<Value, N, detail::index_sequence<Is...>> {
 public:
     //! Type of a value.
     using value_type = Value;
 
-private:
-    //! Helper type to construct range with N tbb::blocked_range<value_type> objects.
-    template<std::size_t>
-    using dim_type_helper = tbb::blocked_range<value_type>;
+    //! Type of a dimension range.
+    using dim_range_type = tbb::blocked_range<value_type>;
 
-public:
-    blocked_rangeNd_impl() = delete;
+    //! Type for the size of a range.
+    using size_type = typename dim_range_type::size_type;
+
+    blocked_nd_range_impl() = delete;
 
     //! Constructs N-dimensional range over N half-open intervals each represented as tbb::blocked_range<Value>.
-    blocked_rangeNd_impl(const dim_type_helper<Is>&... args) : my_dims{ {args...} } {}
+    blocked_nd_range_impl(const indexed_t<dim_range_type, Is>&... args) : my_dims{ {args...} } {}
+
+    blocked_nd_range_impl(const value_type (&size)[N], size_type grainsize = 1) :
```

Comment:
```suggestion
#if __TBB_CLANG_VERSION < 140000
    // On clang prior to version 14.0.0, passing a single braced init list to the constructor of blocked_nd_range<T, 1>
    // matches better on the C array constructor and generates compile-time error because of unexpected size
    // Adding constraints for this constructor to force the compiler to drop it from overload resolution if the size is unexpected
    template <unsigned int M, typename = typename std::enable_if<M == N>::type>
    blocked_nd_range_impl(const value_type (&size)[M], size_type grainsize = 1) :
#else
    blocked_nd_range_impl(const value_type (&size)[N], size_type grainsize = 1) :
#endif
```

I have investigated a bit deeper and it seems like this is a Clang issue and it only takes place on versions prior to 14.0.0. GCC and Microsoft compiler works correct without any changes. 

According to this, I guess we can workaround this only for affected versions of Clang and keep the specification as-is.